### DISCUSSION TITLE
net_listening: how to disable it

### DISCUSSION THREAD

**Discussion by Garito** [wevm/wagmi#2546] [Q&A]
Hi!
Using infura free, I've noticed that the quota was used completely because of errors in the console (http status code 429)
Then I noticed a call (net_listening) that is called regularly and, probably is the one eating the quota
It is possible to disable that call or if not, what is the purpose and why can't be stoped?

Thanks

**Comment by jxom** [comment]
That happens when provider ranking is enabled here: https://wagmi.sh/react/providers/configuring-chains#rank-optional

Do you have it enabled?

Weird that Infura charges for it. Most RPC Providers list it as a free RPC method (it's literally just a ping to see if the server is online lol). Could be worth getting in touch with Infura and asking them if you still want to make use of provider ranking in wagmi!

**Comment by Garito** [comment]
Yeap, I end up noticing it and removed by now
Would be nice if the ranking system discard providers for some configurable time when they return particular error codes (as 429 for infura)

**Comment by Kinco-dev** [comment]
Same error here but with ranking disabled

**Comment by thearyanag** [comment]
hey @jxom , i am facing similar issue with infura

`hey, we are using infura as an rpc, and we checked that the `net_listening` call is taking up most of the usage. 

we use viem, and create transport like this

```
    let transport = fallback(
      [http(chainConfig[chainId])].concat(
        chainId === 8453
          ? [http(chainConfigFallback[chainId])]
          : [webSocket(chainConfigFallback[chainId])]
      ),
      {
        rank: {
          interval: 500,
        },
      }
    );
``` 
where 

```

const chainConfig = {
  1: `https://mainnet.infura.io/v3/${process.env.INFURA_API_KEY}`,
  7777777: "https://rpc.zora.energy",
  8453: `https://base-mainnet.g.alchemy.com/v2/${process.env.ALCHEMY_BASE_API_KEY}`,
  10: `https://optimism-mainnet.infura.io/v3/${process.env.INFURA_API_KEY}`,
  11155111: `https://sepolia.infura.io/v3/${process.env.INFURA_API_KEY}`,
  999999999: "https://sepolia.rpc.zora.energy",
  84532: `https://sepolia.base.org`,
  420: `https://optimism-goerli.infura.io/v3/${process.env.INFURA_API_KEY}`,
  666666666: degen.rpcUrls.default.http[0],
  5112: "https://ham.calderachain.xyz/http",
  16600: "https://rpc-testnet.0g.ai",
  42161: `https://arbitrum-mainnet.infura.io/v3/${process.env.INFURA_API_KEY}`,
};

const chainConfigFallback = {
  84532: `https://sepolia.base.org`,
  666666666: degen.rpcUrls.default.webSocket[0],
  5112: "wss://ham.calderachain.xyz/ws",
  16600: "wss://rpc-testnet.0g.ai/ws",
  42161: `wss://arbitrum-mainnet.infura.io/ws/v3/${process.env.INFURA_API_KEY}`,
};
```

the `net_listening` call is just happening for arbitrum chain`

on discussion with the infura team, they told 
```Depending on what calls you are making on arbitrum, via http or websocket, viem might call  net_listening numerous times. My suggestion is to contact viem support to see how you can lower the frequency of net_listening calls or how to disable it.```

I can disable ranking, but need to know the issue here.

**Reply by jxom** [reply]
Please see https://github.com/wevm/wagmi/discussions/2546#discussioncomment-6179041
